test/cluster/test_tablets_colocation.py

```diff
@@ -440,3 +441,44 @@ async def test_repair_colocated_base_and_view(manager: ManagerClient):
         rows = await cql_server2.run_async(SimpleStatement(f"SELECT * FROM {ks}.tv", consistency_level=ConsistencyLevel.ONE))
         pks = set(row.pk for row in rows)
         assert 1 in pks and 2 in pks
+
+# Verify the default tombstone GC mode for colocated tables is 'timeout',
+# and that altering it to 'repair' is not allowed.
+@pytest.mark.asyncio
+async def test_colocated_tables_gc_mode(manager: ManagerClient):
+    servers = await manager.servers_add(3, auto_rack_dc="dc1")
+    cql = manager.get_cql()
+
+    def check_tombstone_gc_mode_timeout(cql, table):
+        s = list(cql.execute(f"DESC {table}"))[0].create_statement
+        assert "'mode': 'timeout'" in s or "mode" not in s # default is timeout
+
+    async with new_test_keyspace(manager, "WITH replication = {'dc1': 3 }") as ks:
+        async with new_test_table(manager, ks, "pk int PRIMARY KEY, v int") as table:
+            async with new_materialized_view(manager, table, "*", "pk, v", "pk IS NOT NULL AND v IS NOT NULL") as mv:
+                check_tombstone_gc_mode_timeout(cql, mv)
+
+                with pytest.raises(InvalidRequest):
+                    await cql.run_async(f"ALTER TABLE {mv} WITH tombstone_gc = {{'mode': 'repair'}}")
+
+            await cql.run_async(f"CREATE INDEX ON {table}((pk),v)")
+            view_name = cql.execute("SELECT view_name FROM system_schema.views").one().view_name
+            check_tombstone_gc_mode_timeout(cql, f"{ks}.{view_name}")
+
+            with pytest.raises(InvalidRequest):
+                await cql.run_async(f"ALTER TABLE {ks}.{view_name} WITH tombstone_gc = {{'mode': 'repair'}}")
```

Comment:
As far as I know, you can't alter a materialized view via `ALTER TABLE`, you need to use `ALTER MATERIALIZED VIEW`. Please update all the checks in this test so that they check the error message - validating just the type of the error is not enough, the operation can fail due to a different reason.